### PR TITLE
Refactor and updates (Italian notation, support for flat chords, suppport for textScaleFactor, support for metadata)

### PR DESCRIPTION
**Pull request by SamuMazzi**
First of all, thanks for your work. I made some refactor: the code didn't seem to be very readble.
I created a new class for the transposer and added the italian notation.
I added textScaleFactor and scrollPhysics in the renderer.
I added the support for various metadata like chorus, capo, key, artist. Details are in the code. I needed these changings in my code so I've added them but if you have any doubts let me know :)

### CODE REVIEW

**Comment by paurakhsharma** [comment]
Hey, @SamuMazzi thank you so much for this. I will review this PR shortly. Sorry for being late.

File: lib/src/lyrics_renderer.dart [hunk 1]
```diff
@@ -32,46 +33,84 @@ class LyricsRenderer extends StatefulWidget {
...
   late final ScrollController _controller;
+  late TextStyle chorusStyle;
+  late TextStyle capoStyle;
+  bool _isChorus = false;
+
+  @override
+  void initState() {
+    super.initState();
+    chorusStyle = widget.textStyle.copyWith(fontWeight: FontWeight.bold);
+    capoStyle = widget.textStyle.copyWith(fontStyle: FontStyle.italic);
```
Comment (paurakhsharma) [inline lib/src/lyrics_renderer.dart:78]: Make these configurable as well.

**Review by paurakhsharma — CHANGES_REQUESTED** [review]
This looks great. Just a single suggestion. Also, please add trailing commas where possible to format the code nicely.

**Comment by SamuMazzi** [comment]
Did it!
I didn't find where to put trailing commas, I hope it's not a problem...

**Comment by SamuMazzi** [comment]
Did it!

**Review by paurakhsharma — APPROVED** [review]

**Comment by paurakhsharma** [comment]
Thank you so much @SamuMazzi

**Pull request merged**